lib/ruby_indexer/lib/ruby_indexer/rbs_indexer.rb

```diff
@@ -122,7 +128,91 @@ def handle_method(member, owner)
       end
 
       real_owner = member.singleton? ? @index.existing_or_new_singleton_class(owner.name) : owner
-      @index.add(Entry::Method.new(name, file_path, location, location, comments, [], visibility, real_owner))
+      signatures = signatures(member)
+      @index.add(Entry::Method.new(name, file_path, location, location, comments, signatures, visibility, real_owner))
+    end
+
+    sig { params(member: RBS::AST::Members::MethodDefinition).returns(T::Array[Entry::Signature]) }
+    def signatures(member)
+      member.overloads.map do |overload|
+        parameters = process_overload(overload)
+        Entry::Signature.new(parameters)
+      end
+    end
+
+    sig { params(overload: RBS::AST::Members::MethodDefinition::Overload).returns(T::Array[Entry::Parameter]) }
+    def process_overload(overload)
+      function = T.cast(overload.method_type.type, RBS::Types::Function)
+      parameters = parse_arguments(function)
+
+      block = overload.method_type.block
+      parameters << Entry::BlockParameter.anonymous if block&.required
+
+      parameters
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::Parameter]) }
+    def parse_arguments(function)
+      parameters = []
+      parameters += process_required_positionals(function) if function.required_positionals
+      parameters += process_optional_positionals(function) if function.optional_positionals
+      parameters += process_trailing_positionals(function) if function.trailing_positionals
+      parameters += process_rest_positionals(function) if function.rest_positionals
+      parameters += process_required_keywords(function) if function.required_keywords
+      parameters += process_optional_keywords(function) if function.optional_keywords
+      parameters
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::RequiredParameter]) }
+    def process_required_positionals(function)
+      function.required_positionals.map do |param|
+        # Some parameters don't have names, e.g.
+        #   def self.try_convert: [U] (untyped) -> ::Array[U]?
+        name = param.name || :object
+
+        Entry::RequiredParameter.new(name: name)
+      end
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::OptionalParameter]) }
+    def process_optional_positionals(function)
+      function.optional_positionals.map do |param|
+        # Optional positionals may be unnamed, e.g.
+        #  def self.polar: (Numeric, ?Numeric) -> Complex
+        name = param.name || :anon # TODO: what should we call this?
```

Comment:
I wouldn't use `anon` since you cannot actually have an anonymous optional parameter, they must always have some name.

My vote goes for simplicity with `arg0`, `arg1`, `arg2` and so on using a map with index.